src/meta/src/barrier/mod.rs

```diff
@@ -666,36 +672,37 @@ where
             ));
             let mut notifiers = notifiers;
             notifiers.iter_mut().for_each(Notifier::notify_to_send);
-            checkpoint_control.inject(command_ctx.clone(), notifiers);
 
-            self.inject_and_send_err(command_ctx, barrier_complete_tx.clone())
+            checkpoint_control.enqueue_command(command_ctx.clone(), notifiers);
+            self.inject_barrier_and_collect(command_ctx, barrier_complete_tx.clone())
```

Comment:
I think naming `and_collect` makes it more confusing, which sounds like the event loop is blocked by the collecting process.